File: be-cyberclaim/schemas/doctor.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime, date
import uuid

class DoctorBase(BaseModel):
    name: str
    specialization: Optional[str] = None
    bpjs_id: Optional[str] = None
    birth_date: Optional[date] = None
    gender: Optional[str] = None
    telp: Optional[str] = None
    address: Optional[str] = None
    facility_id: uuid.UUID

class DoctorCreate(DoctorBase):
    pass

class DoctorUpdate(BaseModel):
    name: Optional[str] = None
    specialization: Optional[str] = None
    bpjs_id: Optional[str] = None
    birth_date: Optional[date] = None
    gender: Optional[str] = None
    telp: Optional[str] = None
    address: Optional[str] = None
    facility_id: Optional[uuid.UUID] = None
    is_active: Optional[bool] = None

class DoctorResponse(DoctorBase):
    id: uuid.UUID
    is_active: bool
    created_at: datetime
    updated_at: datetime
    facility_name: str
    
    class Config:
        from_attributes = True

File: fe-cyberclaim/types/doctor.ts
// types/doctor.ts
export interface Doctor {
    id?: string;
    facility_id: string;
    name: string;
    specialization: string;
    bpjs_id: string;
    birth_date: string;
    gender: 'L' | 'P';
    telp: string;
    address: string;
    is_active: boolean;
    created_at?: string;
    updated_at?: string;
    facility_name?: string; // Joined field
  }
  
  export interface DoctorFormData {
    facility_id: string;
    name: string;
    specialization: string;
    bpjs_id: string;
    birth_date: string;
    gender: 'L' | 'P';
    telp: string;
    address: string;
    is_active: boolean;
  }